internal/ingress/controller/config/config.go

```diff
@@ -436,6 +436,14 @@ type Configuration struct {
 	// By default this is disabled
 	UseGeoIP2 bool `json:"use-geoip2,omitempty"`
 
+	// GeoIP2AutoReload enables autoreload on geoip2
+	// By default this is disabled
+	GeoIP2AutoReload bool `json:"enable-geoip2-autoreload,omitempty"`
+
+	// GeoIPAutoReloadInterval configures auto_reload interval for geoip2 module
+	// By default this is 5m if GeoIP2AutoReload is true
+	GeoIPAutoReloadInterval string `json:"geoip2-autoreload-interval,omitempty"`
```

Comment:
Thinking here...what if instead of having a bool flag and a time flag, we define just the time flag (like GeoIPAutoReloadInterval) with a default 0 (or empty), and if it is not empty, then we add it?

Also, at some point you may need to do some parsing to match if the time (5m) is acceptable, or if there's something that can break (what happens if I add AutoReloadInterval = 57poXP45 string?"

Maybe, the field can just be "GeoIP2AutoReloadMinutes", then you default to 0, accept an integer and concatenate the integer with the string "m" somewhere else :)

Reply:
I liked this suggestion. Thanks Katz!